staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -38,6 +42,27 @@ type CustomResourceDefinitionSpec struct {
 	// subresources via the CustomResourceSubresources feature gate.
 	// +optional
 	Subresources *CustomResourceSubresources `json:"subresources,omitempty" protobuf:"bytes,6,opt,name=subresources"`
+	// Versions is the list of all supported versions for this resource.
+	// If Version field is provided, this field is optional.
+	// Validation: All versions must use the same validation schema for now. i.e., top
+	// level Validation field is applied to all of these versions.
+	// Order: The order of these versions is used to determine the order in discovery API
+	// (preferred version first).
+	Versions []CustomResourceDefinitionVersion `json:"versions,omitempty" protobuf:"bytes,7,rep,name=versions"`
+}
+
+type CustomResourceDefinitionVersion struct {
+	// Name is the version name, e.g. “v1”, “v2beta1”, etc.
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+	// Served is a flag enabling/disabling this version from being served via REST APIs
+	// Default is True.
```

Comment:
this defaulting is not implemented, is it? It should, i.e. we need `*bool` as type to detect when it is unset.

Reply:
We went back and force between making this optional or not. I don't have a strong opinion either way but lets not make them optional for now as making them optional is a backward compatible change. WDYT?